src/matrix/storage/idb/error.ts

```diff
@@ -18,7 +18,10 @@ limitations under the License.
 import { StorageError } from "../common";
 
 export class IDBError extends StorageError {
-    constructor(message, source, cause) {
+    storeName: string;
+    databaseName: string;
+
+    constructor(message: string, source, cause: DOMException | null) {
```

Comment:
Unfortunately not, since `IDBRequestError` feeds in an `IDBRequest` here.

Reply:
Looks like `IDBRequestError` passes in `request.source` though? This should be of type `IDBIndex | IDBCursor | IDBObjectStore`. So, there is currently a bug where it sometimes says `<unknown store>.<unknown db>` in the error message, I'm guessing that either I messed up and I'm passing the wrong thing in somewhere, or browsers don't always set source.name and source.transaction.db.name. I'm guessing the former. I mainly added the `?` operator out of caution; to ensure the constructor does not crash and hide the real error. It would be good to add some type-safety here nonetheless, while keeping the caution.

While you're at it, at least passing in `IDBCursor` explains the bug, as it does not have a name, but it does have an `objectStore` property. Similarly, `IDBIndex` has an `objectStore` property, but also a `name`, which would be good to include in the error message in the case of an index, like `hydrogen_393920320.timelineEvents.byEventId` or something. So I'd have this constructor accept  `IDBIndex | IDBCursor | IDBObjectStore` for the `source` and then with instanceof narrowing determine in db and object store name for each case in a better way than currently to avoid the aforementioned bug.